analyzers/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldContainAssertion.cs

```diff
@@ -117,8 +122,11 @@ private static bool IsTestIgnored(IMethodSymbol method)
 
         private static bool IsKnownAssertion(ISymbol methodSymbol)
         {
-            var type = knownAssertions.GetValueOrDefault(methodSymbol.Name);
-            return (type != null) && methodSymbol.ContainingType.ConstructedFrom.Is(type);
+            var type = KnownAssertions.GetValueOrDefault(methodSymbol.Name);
+            return type != null
+                ? methodSymbol.ContainingType.ConstructedFrom.Is(type)
+                : methodSymbol.ContainingType is { } containingType
+                  && containingType.DerivesFromAny(KnownAssertionTypes);
```

Comment:
`type != null` can be `is {}`
`ContainingType` cannot be null and we can access it directly.
With that, we can simplify this to an arrow method.
```suggestion
            (KnownAssertions.GetValueOrDefault(methodSymbol.Name) is { } type && methodSymbol.ContainingType.ConstructedFrom.Is(type))
            || methodSymbol.ContainingType.DerivesFromAny(KnownAssertionTypes);
```